emu_sv/state_vector.py

```diff
@@ -41,13 +41,35 @@ def __init__(
         device = "cuda" if gpu and DEVICE_COUNT > 0 else "cpu"
         self.vector = vector.to(dtype=dtype, device=device)
 
+    def _inner(self, other: State) -> torch.Tensor:
```

Comment:
who not move this to `inner` and have a dedicated `_norm` for where the norm is needed?